src/tado-hotwater-zone.js

```diff
@@ -0,0 +1,143 @@
+            .setCharacteristic(Characteristic.SerialNumber, zoneLeader.serialNo)
+            .setCharacteristic(Characteristic.FirmwareRevision, zoneLeader.currentFwVersion);
+    }
+
+    // Updates the hot water service
+    let hotWaterService = hotWaterAccessory.getServiceByUUIDAndSubType(Service.Switch);
+    if (!hotWaterService) {
+        hotWaterService = hotWaterAccessory.addService(Service.Switch);
+    }
+
+    // Stores the hot water service
+    zone.hotWaterService = hotWaterService;
+
+    // Subscribes for changes of the target state characteristic
+    hotWaterService.getCharacteristic(Characteristic.On).on('set', function (value, callback) {
+        // Sets the state to ON
+        if (value === true) {
+            platform.log.debug(zone.id + ' - Switch target state to HEATING');
+            zone.platform.client.setZoneOverlay(platform.home.id, zone.id, 'on', hotWaterService.getCharacteristic(Characteristic.On).value, platform.config.switchToAutoInNextTimeBlock ? 'auto' : 'manual').then(function() {
+                // Updates the state
+                zone.updateState();
+            }, function() {
+                platform.log(zone.id + ' - Failed to switch target state to ON');
+            });
+        }
+
+        // Sets the state to OFF
+        if (value === false) {
+            platform.log.debug(zone.id + ' - Switch target state to AUTO');
+            zone.platform.client.clearZoneOverlay(platform.home.id, zone.id).then(function() {
+                // Updates the state
+                zone.updateState();
+            }, function() {
+                platform.log(zone.id + ' - Failed to switch target state to OFF');
+            });
+        }
+
+        // Performs the callback
+        callback(null);
+    });
+
+    // Sets the interval for the next update
+    setInterval(function() { zone.updateState(); }, zone.platform.config.stateUpdateInterval * 1000);
+
+    // Updates the state initially
+    zone.updateState();
+}
+
+/**
+ * Can be called to update the zone state.
+ */
+TadoHotWaterZone.prototype.updateState = function () {
+    const zone = this;
+    const { Characteristic } = zone.platform;
+
+    // Calls the API to update the state
+    zone.platform.client.getZoneState(zone.platform.home.id, zone.id).then(function(state) {
+        const apiZone = zone.platform.apiZones.find(function(z) { return z.id === zone.id; });
+        apiZone.state = state;
+
+        // Updates the target state
+        zone.hotWaterService.updateCharacteristic(Characteristic.On, state.setting.power === 'ON');
+        
+        zone.platform.log.debug(zone.id + ' - Updated state.');
+        zone.platform.log.debug(zone.id + ' - new state: ' + JSON.stringify(state));
+    }, function() {
+        zone.platform.log(zone.id + ' - Error getting state from API.');
+    });
+}
+
+/**
+ * Can be called to update the zone.
+ */
+TadoHotWaterZone.prototype.updateZone = function (apiZones) {
```

Comment:
This method does not do anything, I would suggest to remove this dead code like this:

```
/**
  * Can be called to update the zone. Not needed for hot water.
  */
TadoHotWaterZone.prototype.updateZone = function (apiZones) { }
```